src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -449,28 +449,29 @@ public void filterSensitiveInformation() {
     }
 
     /**
-     * find a relevant participation for this exercise (relevancy depends on InitializationState)
+     * Find a relevant participation for this exercise (relevancy depends on InitializationState)
      *
      * @param participations the list of available participations
      * @return the found participation, or null, if none exist
      */
     public StudentParticipation findRelevantParticipation(List<StudentParticipation> participations) {
         StudentParticipation relevantParticipation = null;
-        for (StudentParticipation participation : participations) {
-            if (participation.getExercise() != null && participation.getExercise().equals(this)) {
-                if (participation.getInitializationState() == InitializationState.INITIALIZED) {
-                    // InitializationState INITIALIZED is preferred
-                    // => if we find one, we can return immediately
-                    return participation;
-                }
-                else if (participation.getInitializationState() == InitializationState.INACTIVE) {
-                    // InitializationState INACTIVE is also ok
-                    // => if we can't find INITIALIZED, we return that one
-                    relevantParticipation = participation;
-                }
-                else if (participation.getExercise() instanceof ModelingExercise || participation.getExercise() instanceof TextExercise
-                        || participation.getExercise() instanceof FileUploadExercise) {
-                    return participation;
+        if (participations != null) {
```

Comment:
Thank you for your review!

Initially the participations are queried via `participationService.findWithResultsByStudentId()` in `CourseResource.java` and are passed to this method. It could be the case that there is no participation at all for the student and in that case the check prevents a `NullPointerException`. I think it also makes sense for the (unlikely) case that someone reuses the method in a different context :-)